src/main/resources/templates/users/search.html

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org/"
+      xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity5">
+
+<head th:replace="fragments :: head">
+    <meta charset="UTF-8"/>
+    <title></title>
+</head>
+
+<body>
+
+<nav th:replace="fragments :: page-header"></nav>
+
+<div class="container body-content">
+    <h2 th:text="${title}"></h2>
+
+    <td style="vertical-align:middle;">
+        <form method="get" th:action="@{/users/search/}" th:object="${user}">
+            <input type="text" name="searchTerm" id="searchTerm" th:value="${searchTerm}">
+            <button type="submit" class="btn btn-warning">Search</button>
+        </form>
+    </td>
+
+    <p th:text="'Results containing term: ' + ${searchTerm}"></p>
+
+    <table class="table">
+        <tr>
+            <th scope="col">User First Name</th>
+            <th scope="col">User Last Name</th>
+            <th scope="col">Username</th>
+            <th scope="col">Role</th>
+        </tr>
+        <tr th:each="user : ${results}">
+            <td th:text="${user.fname}" style="vertical-align:middle;"></td>
+            <td th:text="${user.lname}" style="vertical-align:middle;"></td>
+            <td th:text="${user.username}" style="vertical-align:middle;"></td>
+            <td th:text="${user.role}" style="vertical-align:middle;"></td>
+            <td style="vertical-align:middle;">
+                <form method="get" th:action="@{/users/update/}" th:object="${user}">
+                    <input type="hidden" th:name="userId" th:value="${user.id}">
+                    <input type="hidden" th:name="pword" th:value="${user.password}">
```

Comment:
Well, I think they are needed because there is validation that password is required.  The way it's coded, it won't work without it.  It is hidden and not being decrypted, so I think it's ok.  We can talk about whether that is a concern, and maybe it can be handled another way.